src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -320,48 +320,29 @@ private static TimeZoneInfo GetLocalTimeZoneFromWin32Data(in TIME_ZONE_INFORMATI
         /// This function will either return a valid TimeZoneInfo instance or
         /// it will throw 'InvalidTimeZoneException' / 'TimeZoneNotFoundException'.
         /// </summary>
-        public static TimeZoneInfo FindSystemTimeZoneById(string id)
+        private static TimeZoneInfoResult TryFindSystemTimeZoneById(string id, out TimeZoneInfo? timeZone, out Exception? e)
```

Comment:
>TryFindSystemTimeZoneById [](http://example.com/codeflow?start=42&length=25)

This is now almost exact to the same one on Unix file. I am seeing the only difference is the check `id.Length > MaxKeyLength` which we can apply on Unix version too. 

I suggest get rid of these two and consolidate them into the one you introduced in the common TimeZoneInfo.cs file.

Reply:
They make different calls:
`TryGetTimeZone(id, false, out timeZone, out e, cachedData);` -- Win32
`TryGetTimeZone(id, false, out timeZone, out e, cachedData, alwaysFallbackToLocalMachine: true);` -- Unix

Reply:
Still worth moving the common code to timezoneinfo.cs and keep here only the different code. what do you think?